File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        return maxDepth(root, 0);
        
    }
    public int maxDepth(TreeNode root, int currDepth){
        if (root == null){
            return currDepth;
        } else {
            currDepth++;
            int lDepth = maxDepth(root.left, currDepth);
            int rDepth = maxDepth(root.right, currDepth);
            return Math.max(lDepth, rDepth);
            // if (lDepth > rDepth){
            //     return lDepth;
            // } 
            // return rDepth;
            
        }
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        q = deque()
        q.append(root)
        
        while (q):
            layer = []
            qLen = len(q)
            for i in range(qLen):
                nxt_elem = q.popleft()
                if nxt_elem:
                    layer.append(nxt_elem.val)
                    q.append(nxt_elem.left)
                    q.append(nxt_elem.right)
            if (layer):
                res.append(layer)
        
        return res
                
                
                
